packages/@glimmer/manager/lib/public/modifier.ts

```diff
@@ -99,41 +99,44 @@ export class CustomModifierManager<O extends Owner, ModifierInstance>
     let delegate = this.getDelegateFor(owner);
 
     let args = argsProxyFor(capturedArgs, 'modifier');
-    let instance: ModifierInstance = delegate.createModifier(definition, args);
+    let modifier: ModifierStateBucket = delegate.createModifier(definition, args);
 
     let tag = createUpdatableTag();
-    let state: CustomModifierState<ModifierInstance>;
+    let state: CustomModifierState<ModifierStateBucket>;
 
     state = {
       tag,
       element,
       delegate,
       args,
-      modifier: instance,
+      definition,
+      modifier,
     };
 
-    registerDestructor(state, () => delegate.destroyModifier(instance, args));
+    registerDestructor(state, () => delegate.destroyModifier(modifier, args));
 
     return state;
   }
 
-  getDebugName(definition: object) {
-    if (typeof definition === 'function') {
-      return definition.name || definition.toString();
-    } else {
-      return '<unknown>';
+  getDebugName({ delegate, definition }: CustomModifierState<ModifierStateBucket>) {
```

Comment:
Something about this isn't right – this signature looks like it's still expecting an instance?

Reply:
iirc, all the getDebugNames are passed the result from create

Reply:
it now receives the definition only